contracts/RenPool.sol

```diff
@@ -70,16 +77,47 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
-
-        renToken.transfer(sender, _amount);
+        require(isLocked == false);
         totalPooled -= _amount;
         balances[sender] -= _amount;
-
+        renToken.transfer(sender, _amount);
+            
         emit RenWithdrawal(sender, _amount);
+
+    }
+
+    function withdrawRequest(uint _amount) external {
+        address sender = msg.sender;
+        uint senderBalance = balances[sender];
+
+        require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
+        require(isLocked == true, "The pool is not locked, please do a regular withdraw");
+        withdrawRequests[sender] = WithdrawRequest(msg.sender, _amount);
+
```

Comment:
Right now it would just overwrite which makes sense I think. But indeed we definitely need a function to cancel a withdraw request. I'll do it on a next PR